mantidproject/mantid
# Add a save option for state the ISIS SANS GUI

Fixes #20294

~~**Do not merge or test before #20229 has been merged**~~

This PR adds a save option for the ISIS SANS state of a particular row.

**To test:**

Please find test data here: smb://olympic/babylon5/Scratch/Anton/GUI_TEST

1. Open the ISIS SANS v2 experimental interface
1. Select `SANS2D` as the instrument (below the data table)
2. For the user file load: USER_SANS2D_154E_2p4_4m_M3_Xpress_8mm_SampleChanger_FRONT.txt
3. For the batch file load: batch_file.csv
4. Go the `Settings>State Diagnostic` tab. You should see a tree view populated with data
5. At the bottom of the tab select a save location and press save.
    * Confirm that a `json` file with the state has been saved out
6. Just place gibberish in the to-save line edit field and press save.
    * Confirm that a warning is shown.


**Release Notes** 
Not required, the GUI is being released in this cycle

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
